Refactored loader.js

Bexxi002 opened the pull request:
[ + ] Added constants `STYLE` and `SCRIPT` for readability
[ ~ ] Moved plugin declarations into an array for easier editing and iteration
[ ~ ] Replaced repeated `queuePlugin` calls with a loop
[ ~ ] Renamed variables
[ ~ ] Standardized `let` / `const` usage and formatting
[ + ] Improved style injection with updated preload support + fallback
[ = ] Behavior unchanged

Bexxi002 committed:
Refactored loader.js

MeowcaTheoRange commented on scripts/furaffinity.net/loader.js:
```diff
@@ -1,89 +1,124 @@
+
+  ["noGalleryPreview", STYLE | SCRIPT],
+  ["showMeTheTags", STYLE | SCRIPT]
+].forEach(([name, type]) => queuePlugin(name, type));
```
I'm not sure I like how you're going about this - this _is_ fancy but I prefer the functional way over the array way.

MeowcaTheoRange replied:
Then again, I will test this.

Bexxi002 replied:
Hey! I went with arrays for better readability / scalability, but id be happy to revert it if you'd prefer that. What do you think about the other changes?

MeowcaTheoRange replied:
I'm doing a little bit of tweaking on my end, and I really like how you did the `STYLE | SCRIPT` thing.
I do also want to note that the plugins were originally separated in a way that makes their dependencies clear - systemMessageOverlay (partially) depends on unwatchDATM, for example.
Tweaking it, I'm for compromising between the array-based format and the functional format, using a function called queuePlugins and giving it an array instead.

Commit:
Change loader.js refactor
- Transform queuePlugin into a function that accepts a list
- Remove queuePlugins logging plugin types
- Change `[...].forEach()` to `queuePlugins([...])`
- Remove old injectStyle comment
- Remove preloading setup - preload does nothing from my testing, though I do think this could be looked into

MeowcaTheoRange commented:
Awwugh I'm not sure how to make a reviewable commit in VS Code so there's the commit. Revert/commit on it if you desire.

Bexxi002 commented:
> Awwugh I'm not sure how to make a reviewable commit in VS Code so there's the commit. Revert/commit on it if you desire.

Thank you!! The tweaks look great, happy I could contribute :D

Commit:
Resolve(?) conflicts

Commit:
Merge branch 'main' into pr/Bexxi002/1-1

MeowcaTheoRange commented:
Thanks for the contribution! I made a post about this:
https://www.furaffinity.net/view/60552223/
